### PR TITLE
ROE-704 Validation for presenter full name characters

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-704

### Change description

Added the otherDataString regex and appended it to the express validation.

Had to put the regex string in the same file; for some reason if you try to export it from elsewhere the validation no longer works.

In the regex string had to remove '\' before the '[' character as sonar flagged it as an unnecessary escape character.

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/validation/presenter.validation.ts [hunk 1]
```diff
@@ -2,7 +2,9 @@ import { body } from "express-validator";
 
 import { ErrorMessages } from "./error.messages";
 
+const VALID_CHARACTERS = /^[-,.:; 0-9A-Z&@$£¥€'"«»''""?!/\\()[\]{}<>*=#%+ÀÁÂÃÄÅĀĂĄÆǼÇĆĈĊČÞĎÐÈÉÊËĒĔĖĘĚĜĞĠĢĤĦÌÍÎÏĨĪĬĮİĴĶĹĻĽĿŁÑŃŅŇŊÒÓÔÕÖØŌŎŐǾŒŔŖŘŚŜŞŠŢŤŦÙÚÛÜŨŪŬŮŰŲŴẀẂẄỲÝŶŸŹŻŽa-zÀÖØſƒǺǿẀẅỲỳ]*$/;
```
Comment (mwestacott) [inline src/validation/presenter.validation.ts:5]: Having this here is probably fine for this PR as it's the first occurrence of its use, but in the next PR, we don't want to keep duplicating this string so we'll have to then move it somewhere central and work out how to get validation working with the imported constant value.

File: src/validation/error.messages.ts [hunk 2]
```diff
@@ -56,5 +56,7 @@ export enum ErrorMessages {
   MAX_PUBLIC_REGISTER_NAME_LENGTH = "Name of register must be 4000 characters or less",
   MAX_PUBLIC_REGISTER_NUMBER_LENGTH = "Entity’s registration number must be 32 characters or less",
   MAX_OCCUPATION_LENGTH = "Occupation must be 100 characters or less",
-  MAX_ROLE_LENGTH = "Role and responsibilities must be 4000 characters or less"
+  MAX_ROLE_LENGTH = "Role and responsibilities must be 4000 characters or less",
+  // Invalid characters
+  INVALID_CHARACTERS = "Full name must only include letters a to z, numbers, and special characters such as hyphens, spaces and apostrophes"
```
Comment (markpit) [inline src/validation/error.messages.ts:61]: I would name this INVALID_CHARACTERS_FULL_NAME as the text relates to the Full name field

Comment (mattch1) [reply src/validation/error.messages.ts:61]: Have changed this, however, it could be applied to other fields.

**Review by markpit — DISMISSED** [review]

File: src/validation/regex/regex.validation.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export const INVALID_CHARACTERS_FULL_NAME = /^[-,.:; 0-9A-Z&@$£¥€'"«»''""?!/\\()[\]{}<>*=#%+ÀÁÂÃÄÅĀĂĄÆǼÇĆĈĊČÞĎÐÈÉÊËĒĔĖĘĚĜĞĠĢĤĦÌÍÎÏĨĪĬĮİĴĶĹĻĽĿŁÑŃŅŇŊÒÓÔÕÖØŌŎŐǾŒŔŖŘŚŜŞŠŢŤŦÙÚÛÜŨŪŬŮŰŲŴẀẂẄỲÝŶŸŹŻŽa-zÀÖØſƒǺǿẀẅỲỳ]*$/;
```
Comment (mwestacott) [inline src/validation/regex/regex.validation.ts:1]: If this has been moved somewhere central to be generic and for reuse then suggest changing constant name to remove _FULL_NAME at the end.

Comment (markpit) [reply src/validation/regex/regex.validation.ts:1]: yes, I missed that, I thought the regex was for valid chars? so should just be VALID_CHARACTERS

Comment (mattch1) [reply src/validation/regex/regex.validation.ts]: More or less said the same above, so changed this.

File: src/validation/presenter.validation.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import { body } from "express-validator";
 
 import { ErrorMessages } from "./error.messages";
+import { VALID_CHARACTERS } from "./regex/regex.validation";
 
 export const presenter = [
-  body("full_name").not().isEmpty({ ignore_whitespace: true }).withMessage(ErrorMessages.FULL_NAME).isLength({ max: 160 }).withMessage(ErrorMessages.MAX_FULL_NAME_LENGTH),
+  body("full_name").not().isEmpty({ ignore_whitespace: true }).withMessage(ErrorMessages.FULL_NAME).isLength({ max: 160 }).withMessage(ErrorMessages.MAX_FULL_NAME_LENGTH).matches(VALID_CHARACTERS).withMessage(ErrorMessages.INVALID_CHARACTERS),
```
Comment (mwestacott) [inline src/validation/presenter.validation.ts]: .matches(VALID_CHARACTERS).withMessage(ErrorMessages.INVALID_CHARACTERS)

doesn't make sense and I think the VALID_CHARACTERS constant should change its name back to INVALID_CHARACTERS.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**